### PR TITLE
Support for new_empty, new_zeros, new_full

### PR DESCRIPTION
**Pull request by aazzolini**

### CODE REVIEW

File: spmd/tensor/ops/tensor_ops.py [hunk 1]
```diff
@@ -59,6 +59,23 @@ def no_shard_prop_rule(op_schema: OpSchema) -> OutputSharding:
...
+    # this op would benefit from backward sharding propagation!
+    # Since we cannot do that yet, just return replicated
+    input = op_schema.args_schema[0]
+    size = torch.Size(cast(Sequence[int], op_schema.args_schema[1]))
+    assert isinstance(input, DTensorSpec)
+
+    return OutputSharding(
+        output_spec=DTensorSpec(
+            mesh=input.mesh,
+            placements=(Replicate(),) * input.mesh.ndim,
```
Comment (wanchaol) [inline spmd/tensor/ops/tensor_ops.py:72]: nit: [Replicate()]?

File: spmd/tensor/ops/tensor_ops.py [hunk 2]
```diff
@@ -59,6 +59,23 @@ def no_shard_prop_rule(op_schema: OpSchema) -> OutputSharding:
     return OutputSharding(tensor_spec)
 
 
+def new_no_arg_prop_rule(op_schema: OpSchema) -> OutputSharding:
```
Comment (wanchaol) [inline spmd/tensor/ops/tensor_ops.py:62]: maybe call it `new_factory_rule`?

**Review by wanchaol — APPROVED** [review]
lgtm, thanks!

File: spmd/tensor/ops/tensor_ops.py (hunk 1 above)
Comment (aazzolini) [reply spmd/tensor/ops/tensor_ops.py:72]: Do we have a preference for list vs. tuple on this one? If so, should we make typing more strict?

Comment (wanchaol) [reply spmd/tensor/ops/tensor_ops.py]: I think it's actually accepting both tuple and list, I don't have a preference on this actually, usually we use List in examples or tests, but tuple should also work

**Pull request merged**